timesketch/api/v1/resources/analysis.py

```diff
@@ -137,25 +179,18 @@ def post(self, sketch_id):
                 HTTP_STATUS_CODE_FORBIDDEN,
                 'Unable to run an analyzer without any data submitted.')
 
-        timeline_id = form.get('timeline_id')
-        if not timeline_id:
+        timeline_ids = form.get('timeline_ids')
+        if not timeline_ids:
             return abort(
                 HTTP_STATUS_CODE_BAD_REQUEST,
                 'Need to provide a timeline ID')
 
-        timeline = Timeline.query.get(timeline_id)
-        if timeline not in sketch.timelines:
-            return abort(
-                HTTP_STATUS_CODE_FORBIDDEN,
-                'Timeline is not part of this sketch')
-
-        search_index = timeline.searchindex
-
-        if not search_index:
-            return abort(
-                HTTP_STATUS_CODE_BAD_REQUEST, (
-                    'No timeline was found, make sure you\'ve got the correct '
-                    'timeline ID or timeline name.'))
+        for timeline_id in timeline_ids:
+            timeline = Timeline.query.get(timeline_id)
+            if timeline not in sketch.timelines:
```

Comment:
There are two approaches here... 

a) just do a list of timelines in a sketch

```
sketch_timelines = [t.id for t in sketch.timelines]
```

And then just compare the timeline_ids to that...

```
timeline_ids = [t for t in timeline_ids if t in sketch_timelines]
```

(or just a check to see if a timeline is not in there)

```
sketch_timelines = set([t.id for t in sketch.timelines])
if not set(timeline_ids).issubset(sketch_timelines):
    abort(....'timeline ID not part of a sketch sentence, bla bla')
```

and then you can get this all done that way.... 

that way you don't have to do a DB query for each of the timelines there, to check if the timeline is not part of the sketch timelines.... 


also if you insist on doing it this way, you would also need a check : 

```
timeline = Timeline.....
if not timeline:
  abort(....
if timeline not in sketch....
```

that is you also assume that `timelien` exists, so if I pass a non-existing timeline ID this will fail, since None object is not in sketch.timelines, etc